openassessment/data.py

```diff
@@ -498,6 +500,111 @@ def collect_ora2_data(cls, course_id):
         ]
         return header, rows
 
+    @classmethod
+    def collect_ora2_summary(cls, course_id):
+        """
+        Query database for aggregated ora2 summary data.
+
+        Args:
+            course_id (string) - the course id of the course whose data we would like to return
+
+        Returns:
+            A tuple containing two lists: headers and data.
+
+            headers is a list containing strings corresponding to the column headers of the data.
+            data is a list of lists, where each sub-list corresponds to a row in the table of all the data
+                for this course.
+
+        """
+
+        items = AssessmentWorkflow.objects.filter(course_id=course_id)
+
+        # need the workflow steps set and sorted here so the data columns line
+        # up with the headers
+        steps = sorted(AssessmentWorkflow.STEPS)
+
+        rows = []
+        for aw in items:
+            statuses = aw.status_details()
+            submission_dict = sub_api.get_submission_and_student(aw.submission_uuid)
+
+            steps_statuses = []
+            peers_graded = 0
+            graded_by_count = 0
+            for step in steps:
```

Comment:
The client's new diagrams mention `current_step`. Is this something we can provide with the available data?

Reply:
We can provide a status from the assessment workflow. Could be a step, or a status like 'done', 'cancelled'. Should be enough. Also means the `is_cancelled` column is redundant and can be removed.

Reply:
Perfect!